lib/streamlit/elements/multiselect.py

```diff
@@ -40,8 +41,44 @@
 from .form import current_form_id
 from .utils import check_callback_rules, check_session_state_rules
 
+T = TypeVar("T")
+
 
 class MultiSelectMixin:
+    @overload
+    def multiselect(
+        self,
+        label: str,
+        options: Sequence[T],
+        default: Optional[Any] = None,
+        format_func: Callable[[Any], Any] = str,
+        key: Optional[Key] = None,
+        help: Optional[str] = None,
+        on_change: Optional[WidgetCallback] = None,
+        args: Optional[WidgetArgs] = None,
+        kwargs: Optional[WidgetKwargs] = None,
+        *,  # keyword-only arguments:
+        disabled: bool = False,
+    ) -> List[T]:
+        ...
+
+    @overload
+    def multiselect(
+        self,
+        label: str,
+        options: OptionSequence,
+        default: Optional[Any] = None,
```

Comment:
Hm, should we change the type of the default argument in the actual implementation to `Optional[Any]` to match? (Or change the type of this argument to match the implementation, but I think I'd prefer the shorter but equivalent type.

Reply:
Good point. And I guess it could be `Optional[T]` in the `options: Sequence[T]` case above, right?

Reply:
Note that the reason I did not change `default` to `Optional[T]` is the perhaps confusing way `T` is bound when it's split into two variables like that. Consider this [mypy playground](https://mypy-play.net/?mypy=latest&python=3.10&gist=30afd5afd9fc8b9118938802539675e3). Setting different types for `options` and `default` will not produce an error, it will just give you a `T` that satisfies both `options` and `default` (often `object`). 

If you still want to have `Optional[T]` after considering this example I'll make the change

Reply:
Ah, okay. Thanks for the clarification